### ISSUE TITLE
The `stdin` and `stdout` options must not target a file path string that is the same.

### ISSUE THREAD

**Issue by adymorz** [sindresorhus/execa#1057]
The following code throws the error message "The `stdin` and `stdout` options must not target a file path string that is the same." using execa 9.1.0:

```js
await execa(
	"npx",
	["ava-tap-json"],
	{
		stdin: {file: "tests/output/tapOutput.txt"},
		stdout: {file: "tests/output/tapOutput.json"},
		buffer: false
	}
);
```

Meanwhile the following code shows the generated JSON on the console:

```js
await execa(
	"npx",
	["ava-tap-json"],
	{
		stdin: {file: "tests/output/tapOutput.txt"},
		stdout: process.stdout,
		buffer: false
	}
);
```

The code before is working with execa 8.0.1 and 9.1.0:

```js
let inputFile = fs.createReadStream("tests/output/tapOutput.txt");
let outputFile = fs.createWriteStream("tests/output/tapOutput.json");
let tapConverter = execa("npx", ["ava-tap-json"], {buffer: false});
inputFile.pipe(tapConverter.stdin);
tapConverter.stdout.pipe(outputFile, {end: false});
await tapConverter;
```

In the documentation I do not see a restriction to use a file for `stdin` and `stout` at the same time.

I will stick with the old code for now. Can you have a look please what is causing the error message mentioned?

**Comment by ehmicky** [comment]
Hi @adymorz,

Thanks for reporting this bug. I created a PR at #1058 to fix it.

It's nice to see that the new `file` feature is greatly simplifying your previous code, that was the intent. :)

**Comment by ehmicky** [comment]
This was released in [`9.2.0`](https://github.com/sindresorhus/execa/releases/tag/v9.2.0).
